pkg/apis/controlplane/types.go

```diff
@@ -246,3 +247,43 @@ type NetworkPolicyList struct {
 	metav1.ListMeta
 	Items []NetworkPolicy
 }
+
+// The category of NetworkPolicy, could be NetworkPolicy, ClusterNetworkPolicy, and AntreaNetworkPolicy.
+type NetworkPolicyCategory string
+
+const (
+	// It represents the K8s NetworkPolicy.
+	K8sNetworkPolicy NetworkPolicyCategory = "K8sNetworkPolicy"
+	// It represents the Antrea ClusterNetworkPolicy.
+	ClusterNetworkPolicy NetworkPolicyCategory = "ClusterNetworkPolicy"
+	// It represents the Antrea NetworkPolicy.
+	AntreaNetworkPolicy NetworkPolicyCategory = "AntreaNetworkPolicy"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// NodeStatsSummary contains stats produced on a Node. It's used by the antrea-agents to report stats to the antrea-controller.
+type NodeStatsSummary struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// The TrafficStats collected from the Node.
+	NetworkPolicies []NetworkPolicyStats
+}
+
+// NetworkPolicyReference identifies a NetworkPolicy across all kinds of NetworkPolicies.
+type NetworkPolicyReference struct {
+	// The category of the NetworkPolicy.
+	Category NetworkPolicyCategory
+	// The Namespace of the NetworkPolicy. It's empty for ClusterNetworkPolicy.
+	Namespace string
+	// The name of the NetworkPolicy.
+	Name string
+}
+
+// NetworkPolicyStats contains the information and traffic stats of a NetworkPolicy.
+type NetworkPolicyStats struct {
+	// The reference of the NetworkPolicy.
+	NetworkPolicy NetworkPolicyReference
```

Comment:
Do you think if we use UID, we can save quite some traffic between Agent and Controller, compared to three strings?

Reply:
Replied here: https://github.com/vmware-tanzu/antrea/pull/1172#discussion_r487954032, will make the change if you agree with it.

Reply:
Because https://github.com/vmware-tanzu/antrea/pull/1258 also introduces the struct NetworkPolicyReference, I'm going to reuse it but setting UID field only for stats report.